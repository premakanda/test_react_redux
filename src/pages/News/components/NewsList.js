import React from 'react'
import PropTypes from 'prop-types'

const NewsList = ({ data }) => {
  return (
    <div className={'news-list'}>
      {data.map(item => {
        return (
          <div key={item.id} className={'news-item'}>
            <h3>{item.title}</h3>
            <p>{item.text}</p>
          </div>
        )
      })}
    </div>
  )
}

NewsList.proptypes = {
  data: PropTypes.shape({
    data: PropTypes.array.isRequired,
    isLoading: PropTypes.bool.isRequired,
    errorMsg: PropTypes.string,
  }).isRequired,
}

export default NewsList
