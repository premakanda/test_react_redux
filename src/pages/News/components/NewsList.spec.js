import React from 'react'
import { shallow } from 'enzyme'
import NewsList from './NewsList'

describe('NewsList', () => {
  const props = {
    data: [
      {
        id: 1,
        title: 'a',
        text: 'b',
      },
      {
        id: 2,
        title: 'a2',
        text: 'b2',
      },
    ],
  }
  const newsList = shallow(<NewsList {...props} />)

  it('renders properly', () => {
    expect(newsList).toMatchSnapshot()
  })

  it('renders 2 news', () => {
    expect(newsList.find('div.news-item')).toHaveLength(2)
  })

  describe('correct render for first news', () => {
    it('title', () => {
      expect(
        newsList
          .find('h3')
          .first()
          .text()
      ).toEqual(props.data[0].title)
    })

    it('info', () => {
      expect(
        newsList
          .find('div.news-item')
          .first()
          .find('p')
          .first()
          .text()
      ).toEqual(props.data[0].text)
    })
  })
})
