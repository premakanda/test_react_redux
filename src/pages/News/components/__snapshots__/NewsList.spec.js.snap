// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsList renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsList
    data={
      Array [
        Object {
          "id": 1,
          "text": "b",
          "title": "a",
        },
        Object {
          "id": 2,
          "text": "b2",
          "title": "a2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="news-item"
        >
          <h3>
            a
          </h3>
          <p>
            b
          </p>
        </div>,
        <div
          className="news-item"
        >
          <h3>
            a2
          </h3>
          <p>
            b2
          </p>
        </div>,
      ],
      "className": "news-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              a
            </h3>,
            <p>
              b
            </p>,
          ],
          "className": "news-item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "a",
            },
            "ref": null,
            "rendered": "a",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "b",
            },
            "ref": null,
            "rendered": "b",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              a2
            </h3>,
            <p>
              b2
            </p>,
          ],
          "className": "news-item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "a2",
            },
            "ref": null,
            "rendered": "a2",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "b2",
            },
            "ref": null,
            "rendered": "b2",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="news-item"
          >
            <h3>
              a
            </h3>
            <p>
              b
            </p>
          </div>,
          <div
            className="news-item"
          >
            <h3>
              a2
            </h3>
            <p>
              b2
            </p>
          </div>,
        ],
        "className": "news-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                a
              </h3>,
              <p>
                b
              </p>,
            ],
            "className": "news-item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "a",
              },
              "ref": null,
              "rendered": "a",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "b",
              },
              "ref": null,
              "rendered": "b",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                a2
              </h3>,
              <p>
                b2
              </p>,
            ],
            "className": "news-item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "a2",
              },
              "ref": null,
              "rendered": "a2",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "b2",
              },
              "ref": null,
              "rendered": "b2",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
